lib/middleware/prototype-admin-routes.js

```diff
@@ -34,4 +34,19 @@ router.post('/password', (req, res) => {
   }
 });
 
+router.get('/reset', (req, res) => {
+  const returnPage = req.query.returnPage || '/';
+  res.render('reset', {
+    returnPage,
+  });
+});
+
+router.all('/reset-session-data', (req, res) => {
+  req.session.data = {};
+  const returnPage = req.body.returnPage || '/';
```

Comment:
Be aware we still need to reject `returnPage` if it's an external URL

Maybe something crude like checking it begins with `/` is enough?

```suggestion
  const returnPage = req.body.returnPage?.startsWith('/')
    ? req.body.returnPage // Local paths only
    : '/';

  req.session.data = {};

```

Reply:
But equally it might not be a string if someone had hand-crafted the POST (or hidden input)

Reply:
committed your suggestion. Feels good enough? Not sure how it could be a non-string?

Reply:
We set `body-parser` `extended: true` in https://github.com/nhsuk/nhsuk-prototype-kit/blob/e60c9845adfe942076d4e50f530ac28031600a86/app.js#L105-L107

Which from memory means non-strings come from:

### POST as array value

```patch
- <input type="hidden" name="returnPage" value="{{ returnPage }}">
+ <input type="hidden" name="returnPage[0]" value="{{ returnPage }}">
```

### POST as object value

```patch
- <input type="hidden" name="returnPage" value="{{ returnPage }}">
+ <input type="hidden" name="returnPage[name]" value="{{ returnPage }}">
```